con-master/src/main/java/ba/codecentric/master/controller/FacilityController.java

```diff
@@ -27,12 +29,21 @@ public FacilityController(FacilityService facilityService, ImageService imageSer
     }
 
     @RequestMapping(value = "/facilities", method = RequestMethod.POST, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
-    public Facility saveFacility(@RequestPart("image") MultipartFile image, @Valid @RequestPart("facility") Facility facility) throws IOException {
+    public Facility saveFacility(@RequestPart("image") MultipartFile image, @Valid @RequestPart("facility") Facility facility) {
 
-        String fileName = imageService.saveImage(image.getInputStream(), image.getOriginalFilename());
-        if (fileName != null) {
-            facility.setImage(fileName);
-            return facilityService.saveFacility(facility);
+        try {
+
+            String fileName = imageService.saveImage(image.getInputStream(), image.getOriginalFilename());
+            if (fileName != null) {
+                facility.setImage(fileName);
+                log.info("Save facility: " + facility.getFacilityName());
+                return facilityService.saveFacility(facility);
+            }
+            log.error("Error while saving facility image");
```

Comment:
Which error we will have here. Describe it more precisely. Don't use generic messages. Error means nothing. If you already know what would happen, describe it.